controllers/runnermanager_test.go

```diff
@@ -191,6 +193,104 @@ var _ = Describe("RunnerManager", func() {
 	})
 
+	It("should delete runner pods, if it is stale", func() {
+		type inputPod struct {
+			spec         *corev1.Pod
+			ip           string
+			deletionTime time.Time
+		}
+		testCases := []struct {
+			name             string
+			inputRunnerPool  *meowsv1alpha1.RunnerPool
+			inputPods        []*inputPod
+			inputRunners     map[string][]*github.Runner
+			expectedPodNames []string // slice of "<Namespace>/<Pod name>"
+		}{
+			{
+				name:            "delete pods",
+				inputRunnerPool: makeRunnerPool("rp1", "test-ns1", "repo1"),
+				inputPods: []*inputPod{
+					{spec: makePod("pod1", "test-ns1", "rp1"), ip: "10.0.0.1"},
+					{spec: makePod("pod2", "test-ns1", "rp1"), ip: "10.0.0.2"},
+				},
+				expectedPodNames: nil,
+			},
+			{
+				name:            "should not delete pods if that is busy",
+				inputRunnerPool: makeRunnerPool("rp1", "test-ns1", "repo1"),
+				inputPods: []*inputPod{
+					{spec: makePod("pod1", "test-ns1", "rp1"), ip: "10.0.0.1"},
+					{spec: makePod("pod2", "test-ns1", "rp1"), ip: "10.0.0.2"},
+				},
+				inputRunners: map[string][]*github.Runner{
+					"repo1": {
+						{Name: "pod1", ID: 1, Online: false, Busy: false, Labels: []string{"test-ns1/rp1"}}, // pod exists
+						{Name: "pod2", ID: 2, Online: true, Busy: true, Labels: []string{"test-ns1/rp1"}},   // pod exists
+					},
+				},
+				expectedPodNames: []string{
+					"test-ns1/pod2",
+				},
+			},
+			{
+				name:            "should not delete pods if a deletion time has already been set",
+				inputRunnerPool: makeRunnerPool("rp1", "test-ns1", "repo1"),
+				inputPods: []*inputPod{
+					{spec: makePod("pod1", "test-ns1", "rp1"), ip: "10.0.0.1"},
+					{spec: makePod("pod2", "test-ns1", "rp1"), ip: "10.0.0.2", deletionTime: time.Now().Add(24 * time.Hour).UTC()},
+				},
+				expectedPodNames: []string{
+					"test-ns1/pod2",
+				},
+			},
+		}
+		for _, tt := range testCases {
+			ttName := fmt.Sprintf("test case name is '%s'", tt.name)
+
+			By("preparing fake clients")
+			runnerPodClient := rc.NewFakeClient()
+			githubClient := github.NewFakeClient("runnermanager-org")
+			runnerManager := NewRunnerManager(ctrl.Log, time.Second, k8sClient, githubClient, runnerPodClient)
+
+			By("preparing pods and runners")
+			for _, inputPod := range tt.inputPods {
+				Expect(k8sClient.Create(ctx, inputPod.spec)).To(Succeed(), ttName)
+				created := &corev1.Pod{}
+				Expect(k8sClient.Get(ctx, types.NamespacedName{Name: inputPod.spec.Name, Namespace: inputPod.spec.Namespace}, created)).To(Succeed(), ttName)
+				created.Status.PodIP = inputPod.ip
+				Expect(k8sClient.Status().Update(ctx, created)).To(Succeed(), ttName)
+				runnerPodClient.SetDeletionTimes(created.Status.PodIP, inputPod.deletionTime)
+			}
+			githubClient.SetRunners(tt.inputRunners)
+
+			By("starting runnerpool manager")
+			tt.inputRunnerPool.Spec.StaleDuration = "10s"
+			runnerManager.StartOrUpdate(tt.inputRunnerPool)
+			time.Sleep(15 * time.Second) // Wait for the pod to stale.
```

Comment:
I don't like to sleep.
Can't you set creationTimestamp manually?

Reply:
Umm, I think this is no problem.

Reply:
I had assumed that the creationTimestamp could not be changed manually.